python/lsst/ts/salobj/csc_commander.py

```diff
@@ -272,6 +285,20 @@ def __init__(
             if name not in frozenset(exclude_commands)
         }
 
+        # Warn of synchronous do_ methods
+        all_members = inspect.getmembers(self, predicate=inspect.ismethod)
+        sync_do_member_names = [
+            member[0]
+            for member in all_members
+            if member[0].startswith("do_")
+            and not inspect.iscoroutinefunction(member[1])
+        ]
+        if sync_do_member_names:
+            warnings.warn(
+                f"Synchronous do_ methods are deprecated: {sorted(sync_do_member_names)}",
```

Comment:
I have a slight preference for use of ','.join, as listing method names in an array (as [m1,m2,..] is a bit brutal.

Reply:
I would prefer to keep this simple, and the [] act as a useful delimiter.